include/ibamr/EnthalpyHierarchyIntegrator.h

```diff
@@ -116,6 +117,13 @@ class EnthalpyHierarchyIntegrator : public PhaseChangeHierarchyIntegrator
      */
     void computeDivergenceVelocitySourceTerm(int Div_U_F_idx, const double new_time) override;
 
+    /*!
+     * \brief Register specific enthalpy variable \f$ h \f$.
+     */
+    virtual void
+    registerSpecificEnthalpyVariable(SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > h_var,
+                                     const bool output_h_var = true);
```

Comment:
```suggestion
                                    bool output_h_var = true);
```

Reply:
Don't `const` qualify in the interface when passing by value. (This is an implementation detail.)

Reply:
Yes. I will correct them in all the phase change classes.

Reply:
One thing that you can check is that you can remove `const` in the header but not in the implementation, and everything will be fine.

It is totally fine to `const` qualify arguments that are passed by value in an implementation, but it is not really meaningful to do it in the interface, because (AFAIK) C++ does not enforce that the implementation also uses the same `const` qualification. E.g., I think that this is valid code:

```
int foo(const int);

int main() {
    foo(10);
    return 0;
}

int foo(int x) {
    return x + 10;
}
```
and so is this:
```
int foo(int);

int main() {
    foo(10);
    return 0;
}

int foo(const int x) {
    return x + 10;
}
```
and this:
```
class Foo
{
public:
    Foo() = default;
    ~Foo() = default;
    int foo(const int);
};

int main() {
    Foo f;
    f.foo(10);
    return 0;
}

int Foo::foo(int x) {
    return x + 10;
}
```
The main point is that the compiler doesn't care if you `const` qualify an argument that is passed by value in a function prototype --- you can revoke its `const`-ness in the implementation.

Reply:
Thanks. I will remove the ```const``` keyword in the prototype and keep it in the implementation even though the ```const``` keyword is redundant in both the interface and the implementation when the function has pass-by-value arguments.